Northstar.CustomServers/mod/scripts/vscripts/item_inventory/sv_item_inventory.gnut

```diff
@@ -141,6 +142,19 @@ void function PlayerInventory_PopInventoryItem( entity player )
 	return
 }
 
+void function PlayerInventory_TakeAllInventoryItem( entity player )
+{
+	array<InventoryItem> playerInventoryStack = file.playerInventoryStacks[ player ]
+	int len = playerInventoryStack.len()
+	for(int i=0;i<len;i++)
+	{
+		playerInventoryStack.pop()
+		player.SetPlayerNetInt( "itemInventoryCount", playerInventoryStack.len() )
+		waitthread PlayerInventory_TakeInventoryItem( player )
```

Comment:
can't there only be one weapon in the OFFHAND_INVENTORY slot? Why is this removed every iteration?

Reply:
you're right,we don't need it now
.use .clear() is much faster.